server/server.go

```diff
@@ -1490,6 +1490,9 @@ func RunServerWithOpts(stanOpts *Options, natsOpts *server.Options) (newServer *
 		if err := s.startNATSServer(); err != nil {
 			return nil, err
 		}
+		if natsOpts.Port == server.RANDOM_PORT {
+			natsOpts.Port = nOpts.Port
```

Comment:
I wouldn't expect the server to change the supplied options even if it's a pointer, wouldn't an accessor function be a bit more user friendly? Or maybe something similar to the nats `ClientURL()`

Reply:
Thanks for your review @ripienaar, I did it this way because the underlying library also does it this way. The URL method would also work for us. Whichever way you think is best suits us, so long as we can have the OS assigned port.

Reply:
Ah, ok, well @kozlovic would be main stakeholder here, I just saw it go past and thought that was a bit odd

Reply:
It is true that this is how NATS Server is doing and many NATS Server tests are using the updated options. However, maybe exposing a new API ClientURL() that would either return the underlying natsServer.ClientURL() or "" if connecting remotely, would be better. @adrian-thurston Would you feel like doing this or I can do it easily (and include a test). Just let me know.

Reply:
Oh I'm sorry I was wrong ... we actually do use the port plainly, so having it come back in the options as an int means we don't need to parse it out of the client url.

Reply:
@adrian-thurston To be sure i understand what you are saying: you would prefer the update of options as opposed to have a new ClientURL() api so that you have direct access to the port for some other use, right?

Reply:
Need to replace this "if" statement with:
```
if natsOpts != nil && natsOpts.Port == server.RANDOM_PORT {
```
because as you can see from the CI runs, this otherwise causes a panic since natsOpts can be nil.
The streaming server accepts "nats options" that can be nil, in which case it uses default NATS options, which in this case is obviously not accessible to caller since caller passed nil. Even when provided, the server makes a clone so that it is owned and can't be changed underneath by the user.

I would also like you to add this test in `server_run_test.go`:
```
func TestServerRandomPort(t *testing.T) {
	natsOpts := natsdTest.DefaultTestOptions
	natsOpts.Port = natsd.RANDOM_PORT
	opts := GetDefaultOptions()
	s := runServerWithOpts(t, opts, &natsOpts)
	defer s.Shutdown()

	if natsOpts.Port == natsd.RANDOM_PORT {
		t.Fatal("port was not updated")
	}
	s.Shutdown()

	// Try with no nats options provided to make sure we don't
	// access natsOpts without checking that it is not nil
	s = runServerWithOpts(t, opts, nil)
	s.Shutdown()
}
```
Thanks!

Reply:
That's right, I mean it doesn't need to come back in the options per se, but just to have it available as an int would be great.

For sure, I'll fix for the natsOpts == nil case, add the test and resubmit.

Thank you for your feedback!